src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -305,7 +305,9 @@ public static TDelegate TryCompileWithoutClosure<TDelegate>(this LambdaExpressio
         public static TDelegate TryCompile<TDelegate>(this LambdaExpressionInfo lambdaExpr)
             where TDelegate : class =>
             TryCompile<TDelegate>(lambdaExpr.Body, lambdaExpr.Parameters,
-                Tools.GetParamExprTypes(lambdaExpr.Parameters), lambdaExpr.Body.GetResultType());
+                Tools.GetParamExprTypes(lambdaExpr.Parameters), 
+                lambdaExpr.Type.GetTypeInfo().GetDeclaredMethod("Invoke").ReturnType,

```

Comment:
Can it be just a `typeof(TDelegate).GetTypeInfo().GetDeclaredMethod("Invoke")`

?

Reply:
Maybe, need to check. But I think I will have no time today